core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataDeleteIndexService.java

```diff
@@ -49,6 +50,19 @@
  */
 public class MetaDataDeleteIndexService extends AbstractComponent {
 
+    public static final int DEFAULT_MAX_TOMBSTONES = 500;
+
+    /**
+     * Setting for the maximum tombstones allowed in the cluster state;
+     * prevents the cluster state size from exploding too large, but it opens the
+     * very unlikely risk that if there are greater than MAX_TOMBSTONES index
+     * deletions while a node was offline, when it comes back online, it will have
+     * missed index deletions that it may need to process.
+     */
+    public static final Setting<Integer> SETTING_MAX_TOMBSTONES = Setting.intSetting("cluster.index.max-tombstones",
```

Comment:
can we call this `cluster.indices.tombstones.size` ?  (we use index when it's a setting which is specific to one index)